dotCMS/src/main/java/com/dotcms/rest/api/v1/index/ESIndexResource.java

```diff
@@ -67,392 +67,353 @@
+        
+        final InitDataObject init = auth(request, response);
+
+        if(indexExists(indexName) ){
+            return Response.status(404).build();
+        }
+        
+        idxApi.delete(indexName);
 
+        String message = "Index:" + indexName + " deleted";
+        
+        sendAdminMessage(message, MessageSeverity.INFO,init.getUser(), 5000);
+        
+        
+        return getIndexStatus(request, response);
+
+    }
     /**
-     * @deprecated Use the Update Settings API of Elasticsearch. See <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-update-settings.html">Update Settings</a>
+     * use modIndex instead, e.g. PUT /api/v1/esindex/{indexName}?action=activate
+     * @param httpServletRequest
+     * @param httpServletResponse
+     * @param params
+     * @return
      */
     @Deprecated
     @PUT
-    @Path("/updatereplica/{params:.*}")
-    public Response updateReplica(@Context HttpServletRequest httpServletRequest, @Context final HttpServletResponse httpServletResponse,@PathParam("params") String params) {
-        try {
-            InitDataObject init=auth(params,httpServletRequest, httpServletResponse);
-            String indexName = this.indexHelper.getIndexNameOrAlias(init.getParamsMap(),"index","alias",this.indexAPI);
-            int replicas = Integer.parseInt(init.getParamsMap().get("replicas"));
-            APILocator.getESIndexAPI().updateReplicas(indexName, replicas);
-
-            return Response.ok().build();
-        } catch(DotDataException dt){
-        	Logger.error(this, dt.getMessage());
-        	throw new BadRequestException(dt, dt.getMessage());
-        } catch (Exception e) {
-            Logger.error(this.getClass(),"Exception trying to update index replicas: " + e.getMessage(), e);
-            return ResponseUtil.mapExceptionResponse(e);
-        }
+    @Path("/activate/{params:.*}")
+    public Response activateIndex(@Context HttpServletRequest httpServletRequest, @Context final HttpServletResponse httpServletResponse, @PathParam("params") String params) throws DotDataException, IOException {
+        InitDataObject init=auth(httpServletRequest,httpServletResponse);
+        String indexName = this.indexHelper.getIndexNameOrAlias(init.getParamsMap(),"index","alias",this.indexAPI);
+        return modIndex(httpServletRequest, httpServletResponse, indexName, IndexAction.ACTIVATE.name());
+    }
+    
+    
+    /**
+     * use modIndex instead, e.g. PUT /api/v1/esindex/{indexName}?action=deactivate
+     * @param httpServletRequest
+     * @param httpServletResponse
+     * @param params
+     * @return
+     */
+    @Deprecated
+    @PUT
+    @Path("/deactivate/{params:.*}")
+    public Response deactivateIndex(@Context HttpServletRequest httpServletRequest, @Context final HttpServletResponse httpServletResponse, @PathParam("params") String params) throws DotDataException, IOException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'httpServletRequest' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)